### ISSUE TITLE
Disallow mismatched sample formats in channels

### ISSUE THREAD

**Issue by Paul-Licameli** [audacity/audacity#5101]
Behavior changes:

- There is now a check and a message in the "Make Stereo Track" context menu item: "Mono tracks must have the same sample rate in order to be combined into a stereo track".  Make a similar check for differences of sample format.
- Analogous check, if opening a 3.3 or earlier file with mismatched formats, and a message like this existing message: "This project cannot be opened because it contains a stereo track with different sample rates. This functionality is no longer supported since the release of Audacity version 3.4"

Some dev task breakdown:

- `WaveTrack::mFormat` is moved to WaveTrackData
- `WaveTrack::ConvertToSampleFormat` needs to assign format only once
- `WaveTrack::FormatConsistencyCheck` analogous to `RateConsistencyCheck`

QA:
- [x] Verify behavior changes above
- [x] Test track sample format conversions
- [x] After changing stereo track format, and splitting into mono tracks, both tracks have the correct format
- [x] Verify changes to a narrower format add dither noise (do the invert and mix track; use the extreme dB Meter Range in Interface preferences and logarithmic dB vertical ruler, to see it)
- [x] Swapping channels or splitting to mono after format changes

**Comment by petersampsonaudacity** [comment]
@Paul-Licameli @vsverchinsky 

This looks like a duplicate of:

 **Stereo tracks should not allow different sample formats in the channels #4752**

**Comment by chinakov** [comment]
I used Spectrogram to verify the dithering part - seems like everything as it should be. Thank you, tested.

**Comment by Paul-Licameli** [comment]
> I used Spectrogram to verify the dithering part - seems like everything as it should be. Thank you, tested.

You mean you expected to see not all of it black, and you did.

**Comment by chinakov** [comment]
Yes, precisely. I saw left-over dithering data by inverting and mixing with the original.

**Comment by Paul-Licameli** [comment]
So actually I should have thought harder about what you really should have tested for.

**Comment by chinakov** [comment]
@Paul-Licameli  - Yes, this is how I understood as well. In the screenshot I've posted earlier, you can see some 'noise', or data, and I assume this is caused by dithering. 

Please correct me if I am wrong.
